libvast/src/concept/parseable/vast/type.cpp

```diff
@@ -45,17 +64,7 @@ bool type_parser::parse(Iterator& f, const Iterator& l, Attribute& a) const {
   static auto attr_list = *(skp >> attr);
   // Basic types
   static auto basic_type_parser
-    = "bool" >> attr_list      ->* to_basic_type<bool_type>
-    | "int" >> attr_list       ->* to_basic_type<integer_type>
-    | "count" >> attr_list     ->* to_basic_type<count_type>
-    | "real" >> attr_list      ->* to_basic_type<real_type>
-    | "duration" >> attr_list  ->* to_basic_type<duration_type>
-    | "time" >> attr_list      ->* to_basic_type<time_type>
-    | "string" >> attr_list    ->* to_basic_type<string_type>
-    | "pattern" >> attr_list   ->* to_basic_type<pattern_type>
-    | "addr" >> attr_list      ->* to_basic_type<address_type>
-    | "subnet" >> attr_list    ->* to_basic_type<subnet_type>
-    ;
+    = (parsers::identifier >> attr_list) ->* to_basic_type;
```

Comment:
I think this can be solved within the parser framework. Let's pair this.

Reply:
What's the status on this?

Reply:
I spoke with @tobim. He's going at it.

Reply:
The type parser has been slightly reworked to be more idiomatic.